src/iOS/Xamarin.iOS/Samples/Network Analysis/ConnectedTrace/ConnectedTrace.cs

```diff
@@ -0,0 +1,423 @@
+                if (feature == null) { return; }
+
+                // Create element with `terminal` for junction feature or with `fractionAlong` for edge feature.
+                UtilityElement element = null;
+
+                // Select default terminal or display possible terminals for the junction feature.
+                UtilityNetworkSource networkSource = _utilityNetwork.Definition.GetNetworkSource(feature.FeatureTable.TableName);
+
+                // Check if the network source is a junction or an edge.
+                if (networkSource.SourceType == UtilityNetworkSourceType.Junction)
+                {
+                    // Get the UtilityAssetGroup from the feature.
+                    string assetGroupFieldName = ((ArcGISFeatureTable)feature.FeatureTable).SubtypeField ?? "ASSETGROUP";
+                    int assetGroupCode = Convert.ToInt32(feature.Attributes[assetGroupFieldName]);
+                    UtilityAssetGroup assetGroup = networkSource?.AssetGroups?.FirstOrDefault(g => g.Code == assetGroupCode);
+
+                    // Get the UtilityAssetType from the feature.
+                    int assetTypeCode = Convert.ToInt32(feature.Attributes["ASSETTYPE"]);
+                    UtilityAssetType assetType = assetGroup?.AssetTypes?.FirstOrDefault(t => t.Code == assetTypeCode);
+
+                    // Get the list of terminals for the feature.
+                    IEnumerable<UtilityTerminal> terminals = assetType?.TerminalConfiguration?.Terminals;
+
+                    // If there is more than one terminal, prompt the user to select a terminal.
+                    if (terminals.Count() > 1)
+                    {
+                        // Ask the user to choose the terminal.
+                        UtilityTerminal terminal = await WaitForTerminal(terminals);
+
+                        // Create a UtilityElement with the terminal.
+                        element = _utilityNetwork.CreateElement(feature, terminal);
+                        _statusLabel.Text = $"terminal: `{terminal?.Name ?? "default"}`";
+                    }
+                    else
+                    {
+                        element = _utilityNetwork.CreateElement(feature, terminals.FirstOrDefault());
+                        _statusLabel.Text = $"terminal: `{element.Terminal?.Name ?? "default"}`";
+                    }
+                }
+                else if (networkSource.SourceType == UtilityNetworkSourceType.Edge)
+                {
+                    element = _utilityNetwork.CreateElement(feature);
+
+                    // Compute how far tapped location is along the edge feature.
+                    if (feature.Geometry is Polyline line)
+                    {
+                        line = GeometryEngine.RemoveZ(line) as Polyline;
+
+                        // Set how far the element is along the edge.
+                        element.FractionAlongEdge = GeometryEngine.FractionAlong(line, e.Location, -1);
```

Comment:
It seems like this controls where on the feature the barrier/ origin is, but in the results, the position of the barrier isn't being considered? Is that a bug or a limitation in the feature? Or are the results correct and we're just not showing where on the feature the exact boundary is?